Spigot-Server-Patches/0669-Add-TamebleDeathMessageEvent.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 29 Jan 2021 22:07:02 -0500
+Subject: [PATCH] Add TamebleDeathMessageEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityTameableAnimal.java b/src/main/java/net/minecraft/server/EntityTameableAnimal.java
+index 9e008d56bb88550f399008095734436a5ab768c7..83667ec893a94b58b17e6572e87dbcac6ed066bc 100644
+--- a/src/main/java/net/minecraft/server/EntityTameableAnimal.java
++++ b/src/main/java/net/minecraft/server/EntityTameableAnimal.java
+@@ -1,5 +1,7 @@
+ package net.minecraft.server;
+ 
++import org.bukkit.event.entity.TamebleDeathMessageEvent; // Paper
++
+ import java.util.Optional;
+ import java.util.UUID;
+ import javax.annotation.Nullable;
+@@ -171,7 +173,12 @@ public abstract class EntityTameableAnimal extends EntityAnimal {
+     @Override
+     public void die(DamageSource damagesource) {
+         if (!this.world.isClientSide && this.world.getGameRules().getBoolean(GameRules.SHOW_DEATH_MESSAGES) && this.getOwner() instanceof EntityPlayer) {
+-            this.getOwner().sendMessage(this.getCombatTracker().getDeathMessage(), SystemUtils.b);
++            // Paper start - TamebleDeathMessageEvent
++            TamebleDeathMessageEvent event = new TamebleDeathMessageEvent((org.bukkit.entity.Tameable) getBukkitEntity(), io.papermc.paper.adventure.PaperAdventure.asAdventure(this.getCombatTracker().getDeathMessage()));
```

Comment:
You will need to add an import for the io.papermc.paper class used here because you are in the `net.minecraft.server` package, and there is already an `io` class in there.

Reply:
So I am importing the paper adventure class?